### DISCUSSION TITLE
Total Watched Time Endpoint?

### DISCUSSION THREAD

**Discussion by MatyasKonig** [vimeo/player.js#704] [Q&A]
Not exactly an issue, but a question (was asked to post here by Vimeo support).

I am trying to figure out a way to get total time user spent watching a particular video. The end goal is to collect the following information about every session: what video the user watches and the total duration they watched. An ideal solution would be to have a getter that would retrieve total seconds watched, but I could not find such an API endpoint. Is there a best practice for getting this information?

Using the documentation and API Demo, I have developed the following ideas, but they don’t seem like the best practice. Are the approaches below the correct way to grab total duration of a session?

* Use the play, seek, pause, and stop event listeners to calculate the approximate time user watched. ‘Stop’ - ‘Start’ would get us the time, and we would subtract ‘seeked’ - ‘seeking’ in case the user was skipping around. The problem here seems that there could be numerous events, which make it hard to distinguish the subsegments that need to be added up.
* Adding cue points every second and add up all the cue points triggered. The issue is that cue points trigger even when the user is seeking through the video. The issue here would be that we would need to somehow capture the number of cue points that were triggered during the seek, and reduce that from the total.

Both ideas seem to be workarounds rather than proper solutions. I am sure many people have solved a similar challenge in the past, and we want to take full advantage of Vimeo Player SDK. What is the most efficient way to get the total duration a user has watched during a particular session?

**Comment by fisherinnovation** [comment]
Thanks for reaching out @MatyasKonig - moving this to the discussion board.

Before I get into my response, please know that we are taking your feedback in for internal discussion to possibly identify a better solution that can be provided for this type of request in the future. 

Looking at your two ideas, the first sounds more sound to me. I would also reccomend looking at the `timeupdate` event (https://github.com/vimeo/player.js#timeupdate) which fires at a short interval during playback. This should really help provide a more up to date and finite value of playback duration. This event combined with what you mentioned, might get you a reasonable solution.

As mentioned, we will disscus a more optimized and easier to use method internally and update this discussion with any progress.

Thanks again.

**Reply by MatyasKonig** [reply]
Thank you for moving this to the right place @fisherinnovation!

I appreciate the detailed response and suggestion. I will try to integrate the `timeupdate` event for now.

I hope that this discussion can bring forward some ideas for better solution to this issue. I think getting the total time watched is a valuable metric and many people would benefit from having access to such data.

Thank you!

**Reply by AreViana** [reply]
Do you get a response? I'm in the same situation right now 😞

**Comment by leonioron** [comment]
use the function _getPlayed()_

[https://github.com/vimeo/player.js/#getplayed-promisearray-error](url)


returns an array containing all intervals watched. At the end you sum all these intervals and divide by the total time of the video
